### PR TITLE
Added post simulation tool to send a datastore table to SQLServer

### LINKED ISSUE #6213: Add a post simulation tool to send a datastore table to SQLServer
This is useful when running APSIM on compute clouds where each nodes writes the important data to a central database.

The tool needs to allow the user to specify one or more DataStore table names to send to the database.

### PR DESCRIPTION
**Pull request by hol353**
Resolves #6213

### CODE REVIEW

File: Models/Main.cs [hunk 1]
```diff
@@ -77,6 +77,11 @@ public static void Run(Options options)
                 else if (options.ListSimulationNames)
                     foreach (string file in files)
                         ListSimulationNames(file, options.SimulationNameRegex);
+                else if (options.ListReferencedFileNames)
```
Comment (hol430) [inline Models/Main.cs:80]: This is related to #6129 -  but I wonder if it's worth (in the long run) having a separate assembly for interaction with .apsimx files - e.g. editing them, querying them etc. This addition doesn't really seem any more egregious than the existing option to list simulation names but it does seem odd having them sit next to all of these options for running simulations - they really serve a different purpose.

File: Models/PostSimulationTools/WriteTableToSqlServer.cs [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    [ValidParent(ParentType=typeof(DataStore))]
+    [Serializable]
+    public class WriteTableToSqlServer : Model, IPostSimulationTool
+    {
+        [Link]
+        private IDataStore dataStore = null;
+
+        /// <summary>Connection string.</summary>
+        [Description("Connection string to connect to database.")]
+        public string ConnectionString { get; set; }
```
Comment (hol430) [inline Models/PostSimulationTools/WriteTableToSqlServer.cs:29]: Is this really something we want to store in the .apsimx file?

File: Models/Main.cs (hunk 1 above)
Comment (hol353) [reply Models/Main.cs:80]: Yep I agree. I suspect we're going to have lots of command line options in the future. Agree with a separate assembly for this stuff. In https://github.com/APSIMInitiative/ApsimX/issues/6129 I was suggesting the same thing. We should probably have a model services assembly for all 'APSIM logic'. In there we could have a class for editing, querying of .apsimx files. 

ListSimulationNames and ListReferencedFileNames are also very slow to run for large files like the wheat validation. All I really need for my cloud work is to have the CLI return the number of simulations in the .apsimx file and then the ability to tell the Models CLI to run simulations 10 through 20 i.e. run a subset. This should be quicker than trying to work out simulation names.

File: Models/PostSimulationTools/WriteTableToSqlServer.cs (hunk 2 above)
Comment (hol353) [reply Models/PostSimulationTools/WriteTableToSqlServer.cs:29]: No we don't. This is a similar to our vault discussion the other day. Where do we want to store our secrets?

**Comment by hol353** [comment]
retest this please jenkins

**Comment by hol430** [comment]
retest this please jenkins

**Comment by hol430** [comment]
retest this please jenkins

**Comment by hol430** [comment]
retest this please jenkins

**Comment by hol430** [comment]
retest this please jenkins

**Comment by hol430** [comment]
retest this please jenkins

**Pull request merged**